### DISCUSSION TITLE
Is there something like onStateChange

### DISCUSSION THREAD

**Discussion by Jan078** [KevinVandy/material-react-table#395] [Q&A]
is it possible to save the entire state at once so I don't have to save everything separately ?

the changelog 0.33 says the onStateChange function has been removed, is there an alternative for this

**Comment by KevinVandy** [comment]
It actually still exists because it is a built in TanStack Table option, but it doesn't work for custom states, like all the MRT states. I removed it from TS so that no one would try it and get frustrated when it didn't work as expected.

**Comment by itshallrun** [comment]
@KevinVandy why is it removed? It seems to be tempting to persist the whole state than all the properties one by one? Or is there any other way to persist the whole state?

**Reply by KevinVandy** [reply]
Only half of the states would actually work, as said above. If MRT gets refactored to follow TanStack Tables new custom features guide, it could work in the future.

**Reply by Nilegfx** [reply]
`If MRT gets refactored to follow TanStack Tables new custom features guide, it could work in the future` 
@KevinVandy do you see this happening any soon?

**Reply by KevinVandy** [reply]
If soon means "maybe in the next 12-18 months", then yeah
